src/main/java/net/library/controller/UserController.java

```diff
@@ -1,38 +1,69 @@
 package net.library.controller;
 
-import jakarta.annotation.PostConstruct;
-import net.library.object.User;
+import jakarta.validation.Valid;
+import net.library.exception.MdcUtils;
+import net.library.model.dto.UserDto;
+import net.library.model.request.UserRequest;
+import net.library.model.response.ApiResponse;
+import net.library.model.response.LibraryResponse;
+import net.library.model.response.UserMapper;
 import net.library.service.UserService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.UUID;
 
-import static net.library.util.HttpUtil.*;
+import static net.library.service.validator.UserValidationService.*;
+import static net.library.util.HttpUtil.USERS;
+import static net.library.util.Utils.deleteSpacesHyphens;
 
 @RestController
 @RequestMapping(USERS)
 public class UserController {
 
-    @Autowired
     private final UserService service;
 
     public UserController(final UserService service) {
         this.service = service;
     }
 
-    @PostConstruct
-    public void initUserList() {
-        service.initUserLst();
+    @GetMapping
+    public ResponseEntity<ApiResponse<List<UserDto>>> getAllUsers() {
+
+        return new ResponseEntity<>(new ApiResponse<>(MdcUtils.getCid(), UserMapper.toDto(service.getAllUsers())), HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<LibraryResponse> addUser(@Valid @RequestBody UserRequest userRequest) {
+        final var refinedMsisdn = deleteSpacesHyphens(userRequest.getPhoneNumber());
+        lengthValidator(refinedMsisdn, 9, 12);
```

Comment:
It is better to handle the bulk of validation using @Valid and Bean Validation annotations in UserRequest. It keeps the code clean, enforces consistency, and leverages Spring Boot's automatic validation mechanism. Only use additional validations in the controller or service layer if absolutely necessary (e.g., for complex business rules or database-related checks).